core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -123,12 +133,30 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf:SparkConf = null
+
   override def open(): BlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(fos)
+    val isEncryptedShuffle = sparkConf.getBoolean("spark.encrypted.shuffle", false)
```

Comment:
@CodingCat If we allow BlockObjectWriter#sparkConf is null, some configurations like "spark.encrypted.shuffle" and so on  can not be got.